### ISSUE TITLE
Document the torch.nn.parallel.scatter_gather.gather function

### ISSUE THREAD

**Issue by svekars** [pytorch/pytorch#127899]
Document the `torch.nn.parallel.scatter_gather.gather` function located in [https://github.com/pytorch/pytorch/blob/main/torch/onnx/symbolic_opset9.py#L5416](https://github.com/pytorch/pytorch/blob/main/torch/onnx/symbolic_opset9.py#L5416).
Follow the [Docstring Guidelines](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines).

cc @brycebortree

**Comment by ahoblitz** [comment]
Unassigning while seeking clarification on whether this is for `torch.nn.parallel.scatter_gather.gather` or `torch.onnx.symbolic_opset9.log2`

**Comment by IvanLauLinTiong** [comment]
Hi @svekars , the issue title is documenting about `torch.nn.parallel.scatter_gather.gather` function, so shouldn't the function located in https://github.com/pytorch/pytorch/blob/main/torch/nn/parallel/scatter_gather.py#L86 ?  However, the link you shared above is different.

**Comment by svekars** [comment]
You are correct - apologies for the typo. I also believe that it is already documented at least partially: https://github.com/pytorch/pytorch/blob/main/torch/nn/parallel/scatter_gather.py#L86 - if you have any improvements feel free to raise a PR, otherwise we can close this.
